projects/components/src/filtering/filter/parser/types/abstract-filter-parser.ts

```diff
@@ -0,0 +1,94 @@
+import { FilterAttribute } from '../../filter-attribute';
+import { FilterAttributeType } from '../../filter-attribute-type';
+import { FilterOperator, fromUrlFilterOperator, toUrlFilterOperator, UrlFilterOperator } from '../../filter-operators';
+import { ParsedFilter, SplitFilter } from '../parsed-filter';
+
+export abstract class AbstractFilterParser<TValue> {
+  public abstract supportedAttributeTypes(): FilterAttributeType[];
+  public abstract supportedOperators(): FilterOperator[];
+
+  protected abstract parseValueString(
+    attribute: FilterAttribute,
+    splitFilter: SplitFilter<FilterOperator>
+  ): TValue | undefined;
+
+  protected parseKeyString(attribute: FilterAttribute, splitFilter: SplitFilter<FilterOperator>): string | undefined {
+    return attribute.displayName.toLowerCase() !== splitFilter.lhs.toLowerCase() ? undefined : attribute.name;
+  }
+
+  public parseFilterString(attribute: FilterAttribute, filterString: string): ParsedFilter<TValue> | undefined {
+    const splitFilter: SplitFilter<FilterOperator> | undefined = AbstractFilterParser.splitFilterStringByOperator(
+      this.supportedOperators(),
+      filterString,
+      true
+    );
+
+    return splitFilter !== undefined ? this.parseSplitFilter(attribute, splitFilter) : undefined;
+  }
+
+  public parseUrlFilterString(attribute: FilterAttribute, urlFilterString: string): ParsedFilter<TValue> | undefined {
+    const splitUrlFilter: SplitFilter<UrlFilterOperator> | undefined = AbstractFilterParser.splitFilterStringByOperator(
+      this.supportedOperators().map(toUrlFilterOperator),
+      decodeURIComponent(urlFilterString),
+      false
+    );
+
+    return splitUrlFilter !== undefined
+      ? this.parseSplitFilter(attribute, {
+          lhs: attribute.displayName,
+          operator: fromUrlFilterOperator(splitUrlFilter.operator),
+          rhs: splitUrlFilter.rhs
+        })
+      : undefined;
+  }
+
+  private parseSplitFilter(
+    attribute: FilterAttribute,
+    splitFilter: SplitFilter<FilterOperator>
+  ): ParsedFilter<TValue> | undefined {
+    const parsedKey = this.parseKeyString(attribute, splitFilter);
+
+    if (parsedKey === undefined) {
+      // Unable to parse attribute from lhs
+      return undefined;
+    }
+
+    const parsedValue = this.parseValueString(attribute, splitFilter);
+
+    if (parsedValue === undefined) {
+      // Unable to parse value from rhs
+      return undefined;
+    }
+
+    // Successfully parsed URL filter
+
+    return {
+      field: attribute.name,
+      operator: splitFilter.operator,
+      value: parsedValue
+    };
+  }
+
+  public static splitFilterStringByOperator<TOperator extends string>(
```

Comment:
Why is this public static? Is it because we can't inject here but want to share? Can we either switch to injection to share it, or move to a stand alone function?

Reply:
Yes, this is used in the follow up PR for the new dot syntax. We need to split in order to discover the operator and know which parser to use.

Will change to export const instead of static.